Koans/AboutThreads.cs

```diff
@@ -0,0 +1,158 @@
+            Thread.CurrentThread.Name = "MainThread";
+
+            var result = string.Empty;
+
+            //Create a new thread object with anonymous function
+            Thread otherThread = new Thread(() =>
+            {
+                Thread.CurrentThread.Name = "OtherThread";
+                //Block execution path of the current thread
+                Thread.Sleep(1000);
+                result += Thread.CurrentThread.Name;
+            });
+            //Run the thread   
+            otherThread.Start();
+            otherThread.Join();
+
+            result += Thread.CurrentThread.Name;
+            Assert.Equal(FILL_ME_IN, result);
+        }
+
+        [Step(3)]
+        public void OrderingThreadsPractice()
+        {
+            string result = string.Empty;
+            const string HELLO = nameof(HELLO);
+            const string WORLD = nameof(WORLD);
+
+            Thread thread1 = new Thread(() => { result += HELLO; });
+            Thread thread2 = new Thread(() => { result += WORLD; });
+
+            //start threads so that pass this exercise 
+            thread1.Start();
+            thread1.Join();
+            thread2.Start();
+            thread2.Join();
+
+            Assert.Equal(FILL_ME_IN, result);
+        }
+
+        [Step(4)]
+        public void PassArgumentsToThreads()
+        {
+            int result = default(int);
+
+            Thread thread = new Thread((factor) => { result = (int)factor * 5; });
+            thread.Start(100);
+            thread.Join();
+
+            Assert.Equal(FILL_ME_IN, result);
+        }
+
+
+        [Step(5)]
+        public void LockStatement()
+        {
+            //When two or more thread access shared data or resources
+            //in a way that can lead to unexpected or incorrect behavior
+            //it can cause Race Condition 
+            //
+            //Locking is one way to prevent race conditions by allowing only
+            //one thread to access a shared resource at a time
+
+            object lockObject = new object();
+
+            //Its shared resource that the Sum method access it 
+            //from two different threads
+            int SumResult = 0;
+
+            //We have to do this several times because
+            //the result is not predictable
+            for (int i = 0; i < 5; i++)
+            {
+                // Create two threads that call Sum simultaneously
+                Thread thread1 = new Thread(() => Sum(30));
+                Thread thread2 = new Thread(() => Sum(30));
+                SumResult = 0;
+                thread1.Start();
+                thread2.Start();
+
+                // Wait for the threads to finish
+                thread1.Join();
+                thread2.Join();
+
+                Assert.Equal(60, SumResult);
+            }
+
+            void Sum(int length)
+            {
+                for (int i = 0; i < length; i++)
+                {
+                    Thread.Sleep(100);
```

Comment:
I guess it would be helpful to remind the learner with a comment to remove this sleep statement after they are done with this koan. Otherwise all other koans will take longer to run. Luckily there are now other koans after this one right now.